core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -170,11 +162,8 @@ class JDBCSourceAsColumnarStore(override val connProperties: ConnectionPropertie
   def closeConnection(conn: Option[Connection]): Unit = {
     conn match {
       case Some(conn) if !conn.isClosed =>
-        conn match {
-          case ConnectionType.Embedded =>
-            // conn commit removed txState from the conn context.
-            conn.commit()
-            conn.close()
+        connectionType match {
+          case ConnectionType.Embedded => // Do nothing
```

Comment:
As this is match on connectionType and in case of getConnection we in some cases return PoolConnection, we need to close in that case.
So, may be here findout if the connection is instance of [EmbedConnection] and then exclude close method.